test/lit/passes/optimize-instructions.wast

```diff
@@ -7811,6 +7811,44 @@
       )
     )
   )
+  ;; CHECK:      (func $xor-of-identical-after-fallthrough (param $x i32)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.const 0)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (i32.xor
+  ;; CHECK-NEXT:    (if (result i32)
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:     (return)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (local.get $x)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $xor-of-identical-after-fallthrough (param $x i32)
```

Comment:
This doesn't look like it has anything to do with GC. Is this connected to the `combineOr` change somehow?

Reply:
+1 on splitting non-GC changes into a separate PR